sklearn/cross_decomposition/pls_.py

```diff
@@ -293,6 +293,11 @@ def fit(self, X, Y):
             # 1) weights estimation (inner loop)
             # -----------------------------------
             if self.algorithm == "nipals":
+                # Replace columns that are all close to zero with zeros
+                Y_eps = np.finfo(Yk.dtype).eps
+                Yk_mask = np.all(np.abs(Yk) < 1e3 * Y_eps, axis=0)
```

Comment:
`1e3 * Y_eps` seems large (intuitively). Wouldn't  `1e2 * Y_eps` or `10 * Y_eps` be enough?

Reply:
Looks like `10 * Y_eps` is good enough.